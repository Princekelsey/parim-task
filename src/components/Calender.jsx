import React from "react";
import { useSelector } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectEvents } from "../redux/events/eventsSelectors";
import Event from "./Event";
import featureData from "./data";

const Calender = () => {
  const { eventsList } = useSelector(
    createStructuredSelector({
      eventsList: selectEvents,
    })
  );
  return (
    <section className="features-section">
      <h1 className="heading-primary center-text">
        <span className="heading-primary--main">Events Pro</span>
        <span className="heading-primary--sub">events at a glance</span>
      </h1>
      <div
        style={{ display: "flex", justifyContent: "center" }}
        className="mb-sm"
      >
        <a href="#tours-section" className="btn btn--white btn--animated">
          previous week
        </a>
        <a href="#tours-section" className="btn btn--white btn--animated ml-sm">
          next week
        </a>
      </div>
      <div className="center-text mb-sm">
        <input
          class="styled-checkbox"
          id="styled-checkbox-2"
          type="checkbox"
          value="value2"
          // checked
        />
        <label for="styled-checkbox-2">first day of the week</label>
      </div>
      <div className="row">
        {eventsList.map((data) => (
          <Event date={data.date} key={data.date} event={data.event} />
        ))}
      </div>
    </section>
  );
};

export default Calender;
