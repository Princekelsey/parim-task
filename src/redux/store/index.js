import { createStore, applyMiddleware } from "redux";
import logger from "redux-logger";
import createSagaMiddleware from "redux-saga";
import rootReducer from "../rootReducer";
import rootSaga from "../rootSaga";

// initailize saga middleware
const sagaMiddleware = createSagaMiddleware();

// all middlewares
const middleWares = [sagaMiddleware];

// check if development mode and use logger
if (process.env.NODE_ENV === "development") {
  middleWares.push(logger);
}

// redux store
const store = createStore(rootReducer, applyMiddleware(...middleWares));

// run saga middleware
sagaMiddleware.run(rootSaga);

export default store;
