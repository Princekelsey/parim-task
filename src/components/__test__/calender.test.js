import React from "react";
import ReactDOM from "react-dom";
import Calender from "../Calender";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../redux/store";

afterEach(cleanup);

it("Calender component renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <Provider store={store}>
      <Calender />
    </Provider>,
    div
  );
});

it("renders calender correctly", () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <Calender />
    </Provider>
  );
  expect(getByTestId("calender"));
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Calender />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
