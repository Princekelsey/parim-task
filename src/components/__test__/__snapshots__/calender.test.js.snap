// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="features-section"
  data-testid="calender"
>
  <h1
    className="heading-primary center-text"
  >
    <span
      className="heading-primary--main"
    >
      Events Pro
    </span>
    <span
      className="heading-primary--sub"
    >
      events at a glance
    </span>
  </h1>
  <div
    className="mb-sm"
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-around",
      }
    }
  >
    <button
      className="btn btn--white btn--animated"
      onClick={[Function]}
    >
      <svg
        className="mr-sm"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#55c57a",
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
        />
      </svg>
      previous week
    </button>
    <button
      className="btn btn--white btn--animated ml-sm"
      onClick={[Function]}
    >
      next week
      <svg
        className="ml-sm"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#55c57a",
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
        />
      </svg>
    </button>
  </div>
  <div
    className="center-text mb-sm"
  >
    <input
      checked={false}
      className="styled-checkbox"
      id="styled-checkbox-2"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="styled-checkbox-2"
    >
      first day of the week
    </label>
    <button
      className="btn-text ml-sm"
      onClick={[Function]}
    >
      Reset
    </button>
  </div>
  <div
    className="row"
  >
    <h1
      className="heading-primary center-text"
    >
      Fetching Data &gt;&gt;&gt;
    </h1>
  </div>
</section>
`;
